### PR TITLE
filter for dataset id

### PR DESCRIPTION
**Pull request by brijeshnec**

### CODE REVIEW

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/terms/QQueryTerm.java [hunk 1]
```diff
@@ -939,6 +939,22 @@ private int commonWherePart(String[] attribPath, String[] subAttribPath, String
 			if (!current.getOperator().isEmpty()) {
 				sql.append(" CASE WHEN ");
 				sql.append(currentSqlAttrib);
+				sql.append(" #>'{");
+				sql.append(NGSIConstants.JSON_LD_ID);
+				sql.append("}' ");
+				dollarCount = applyOperator(sql, dollarCount, tuple, false);
+				sql.append(" THEN true");
+//				sql.append(currentSqlAttrib);
```
Comment (amitrnec) [inline Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/terms/QQueryTerm.java]: Remove commented! code

File: QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/repository/QueryDAO.java [hunk 2]
```diff
@@ -800,7 +800,22 @@ public Uni<Tuple2<Map<String, Map<String, Object>>, List<String>>> queryForEntit
...
 				dollar = attrsQuery.toSqlConstructEntity(query, tuple, dollar);
 			} else {
-				query.append("ENTITY.ENTITY");
+				if(qQuery!=null && qQuery.getAttribute().contains(NGSIConstants.QUERY_PARAMETER_DATA_SET_ID)){
+					query.append("""
+                            (jsonb_strip_nulls(jsonb_build_object(
+                            '@id', entity -> '@id',
+                            '@type', entity ->'@type',
+                            '@context', entity ->'@context',
+                            'https://uri.etsi.org/ngsi-ld/scope', entity ->'https://uri.etsi.org/ngsi-ld/scope',
```
Comment (amitrnec) [inline QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/repository/QueryDAO.java]: It have already been added in NGSIContants class so please take it from this class.

File: QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/repository/QueryDAO.java [hunk 3]
```diff
@@ -800,7 +800,22 @@ public Uni<Tuple2<Map<String, Map<String, Object>>, List<String>>> queryForEntit
...
+                            'https://uri.etsi.org/ngsi-ld/createdAt', entity ->'https://uri.etsi.org/ngsi-ld/createdAt',
```
Comment (amitrnec) [inline QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/repository/QueryDAO.java]: "https://uri.etsi.org/ngsi-ld/createdAt" so please take it from NGSIContants class.

File: QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/repository/QueryDAO.java [hunk 4]
```diff
@@ -800,7 +800,22 @@ public Uni<Tuple2<Map<String, Map<String, Object>>, List<String>>> queryForEntit
...
+                            'https://uri.etsi.org/ngsi-ld/modifiedAt', entity ->'https://uri.etsi.org/ngsi-ld/modifiedAt',
```
Comment (amitrnec) [inline QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/repository/QueryDAO.java]: "https://uri.etsi.org/ngsi-ld/modifiedAt" please take it from NGSIContants class

File: QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/repository/QueryDAO.java [hunk 5]
```diff
@@ -800,7 +800,22 @@ public Uni<Tuple2<Map<String, Map<String, Object>>, List<String>>> queryForEntit
...
+                            '%s',
+                            entity ->'%s'
+                            )))""".formatted(qQuery.getAllAttibs().iterator().next().split("\\.datasetId")[0],
```
Comment (amitrnec) [inline QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/repository/QueryDAO.java]: please validate this case without pass the datasetId in parameterized query

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/terms/QQueryTerm.java (hunk 1 above)
Comment (vivekNEC) [reply Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/terms/QQueryTerm.java]: done

File: QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/repository/QueryDAO.java (hunk 2 above)
Comment (vivekNEC) [reply QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/repository/QueryDAO.java]: done

File: QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/repository/QueryDAO.java (hunk 3 above)
Comment (vivekNEC) [reply QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/repository/QueryDAO.java]: done

File: QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/repository/QueryDAO.java (hunk 4 above)
Comment (vivekNEC) [reply QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/repository/QueryDAO.java]: done

**Review by amitrnec — APPROVED** [review]

**Pull request merged**